### PR TITLE
Add arguments when creating Driver for tests so that tests can run

### PR DESCRIPTION
**Pull request by davidungar**
Ensure the diagnostics are not colored, and that the frontend jobs can find the sdk.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci please test

**Review by owenv — COMMENTED** [review]
I think the idea is good, but I'm wondering if we should make this opt-in so there's an indication that it's happening when writing/debugging tests. As far as I know it should only be required for the relatively small number of tests that are running jobs.

File: Tests/SwiftDriverTests/Helpers/DriverExtensions.swift [hunk 1]
```diff
@@ -26,10 +27,52 @@ extension Driver {
...
+    try self.init(
+      args: augment(args: args),
+      env: env,
+      diagnosticsEngine: diagnosticsEngine,
+      fileSystem: fileSystem,
+      executor: executor)
   }
 }
+
+private func augment(args: [String]) throws -> [String] {
```
Comment (CodaFi) [inline Tests/SwiftDriverTests/Helpers/DriverExtensions.swift:39]: Nit: Perhaps a more descriptive name than 'augment(args:)' could be used here.

File: Tests/SwiftDriverTests/Helpers/DriverExtensions.swift [hunk 2]
```diff
@@ -26,10 +27,52 @@ extension Driver {
...
+
+  // Color codes in diagnostics cause mismatches
+  if !args.contains("-color-diagnostics") && !args.contains("-no-color-diagnostics") {
+    augmentedArgs.insert("-no-color-diagnostics", at: extraArgsIndex)
+  }
+
+  // The frontend fails to load the standard library because it cannot
+  // find it relative to the execution path used by the Swift Driver.
+  // So, pass in the sdk path explicitly.
+  if let sdkPath = try cachedSDKPath.get(), !args.contains("-sdk") {
```
Comment (CodaFi) [inline Tests/SwiftDriverTests/Helpers/DriverExtensions.swift:51]: We don't search for the SDK except in immediate mode. The expectation is that users will provide `SDKROOT` in the environment or invoke `swiftc` via `xcrun` so the environment is configured for them. I recommend we provide `SDKROOT` in the testing environment until we land in a toolchain.

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 3]
```diff
@@ -81,7 +81,9 @@ final class SwiftDriverTests: XCTestCase {
     XCTAssertNoThrow(try Driver(args: ["swift", ".foo"]))
     XCTAssertNoThrow(try Driver(args: ["swift", "/foo"]))
 
-    XCTAssertThrowsError(try Driver(args: ["swift", "foo"]))
+    XCTAssertThrowsError(try Driver(args: ["swift", "foo",
```
[the remark this answers is not in the document]
Comment (davidungar) [reply Tests/SwiftDriverTests/SwiftDriverTests.swift:84]: Yes, I think you are right.

File: Tests/SwiftDriverTests/Helpers/DriverExtensions.swift (hunk 1 above)
Comment (davidungar) [reply Tests/SwiftDriverTests/Helpers/DriverExtensions.swift:39]: I'll try to come up with something; it's a good point.

File: Tests/SwiftDriverTests/Helpers/DriverExtensions.swift (hunk 2 above)
Comment (davidungar) [reply Tests/SwiftDriverTests/Helpers/DriverExtensions.swift:51]: Yes, that might be better.

**Comment by davidungar** [comment]
I'll rework this PR--thanks all for your ideas!

**Comment by davidungar** [comment]
Closing this PR in favor of https://github.com/apple/swift-driver/pull/454 and https://github.com/apple/swift-driver/pull/453

**Pull request closed without merging**